### PR TITLE
worked on RouteGuards and implemented them for login, signup

### PR DESCRIPTION
**Pull request by MarcDAFrame**
app/index.ts shows how they are used (just like the regular Route components from react-router)

### CODE REVIEW

**Comment by MarcDAFrame** [comment]
take a look at `src/forms/Base.ts` to see how the dto objects to determine the form errors

take a look at `src/forms/Auth.ts` to see how a BaseForm is implemented, passing the validator through super

take a look at `src/stores/AuthStore.ts` to view how the login system / state is managed

take a look at `src/containers/Auth/index.tsx` to see how material ui is implemented, and the component Input uses forms to feed errors, values, and changes to the form objects.

**Comment by MarcDAFrame** [comment]
was looking for the statusCode in the body not the response object

File: src/app/stores/AuthStore.ts [hunk 1]
```diff
@@ -1,12 +1,76 @@
...
 
    @observable state: 'loading'|'loggedin'|'unauthed'
+
+
+   @observable signupForm: SignupFormModel
+   @observable loginForm: LoginFormModel
+
    constructor (private rootStore: RootStore){
       this.state = 'unauthed'
+      console.log("OY")
```
Comment (saidghamra) [inline src/app/stores/AuthStore.ts:16]: Remove console.log

File: src/app/stores/AuthStore.ts [hunk 2]
```diff
@@ -1,12 +1,76 @@
...
+   }
+
+   loadUser = async () => {
+      this.state = 'loading'
+
+      try{
+         const response = await fetch('/api/user', {
+            method: 'GET',
+            headers: {
+               'Content-Type' : 'application/json'
```
Comment (saidghamra) [inline src/app/stores/AuthStore.ts:29]: Does react have global interceptors like Angular? If possible, we can intercept every API request and add the content-type header to it

File: src/app/forms/Auth.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { LoginUserDto, CreateUserDto } from './validators';
+
+interface signup_i {
+    fullName: string
+    email: string
+    password: string
+}
+export class SignupFormModel extends BaseFormModel<signup_i, signup_i>{
+    constructor() {
+        console.log("hello")
```
Comment (saidghamra) [inline src/app/forms/Auth.ts:11]: Remove console.log

**Comment by saidghamra** [comment]
Will get this into the develop branch and remove console.logs later

File: src/app/stores/AuthStore.ts (hunk 2 above)
Comment (MarcDAFrame) [reply src/app/stores/AuthStore.ts:29]: we can add this in the future.

https://www.npmjs.com/package/fetch-intercept

for now I'll continue this way

File: src/app/forms/Auth.ts (hunk 3 above)
Comment (MarcDAFrame) [reply src/app/forms/Auth.ts:11]: done!

**Pull request merged**